common/src/state/mod.rs

```diff
@@ -742,6 +742,12 @@ impl State {
         fs::write(path, serialized)?;
         Ok(())
     }
+
+    pub fn get_json(&self) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
unless the error is going to get logged or handled, this function may as well return a String and use `unwrap_or_default()`.